contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -868,16 +813,25 @@ contract RibbonTreasuryVault is
      *         This allows all the users to withdraw if the next option is malicious.
      */
     function commitAndClose() external nonReentrant {
+        uint256 stableBalance = IERC20(USDC).balanceOf(address(this));
+
+        // In case chargeAndDistribute was not called last round, call
+        // the function to conclude last round's performance fee and distribution
+        if (stableBalance > 0) {
+            _chargeAndDistribute();
+        }
```

Comment:
We should generally do token transfers at the end of the function

Reply:
See: https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html